packages/backend/src/core/discovery/utils/diffToMessages.ts

```diff
@@ -1,28 +1,85 @@
 import { MAX_MESSAGE_LENGTH } from '../../../peripherals/discord/DiscordClient'
+import { FieldDiff } from './diffContracts'
 import { DiscoveryDiff } from './diffDiscovery'
 
 export function diffToMessages(name: string, diffs: DiscoveryDiff[]): string[] {
-  const header = `${wrapBoldAndItalic(name)} | detected changes`
-  const messages: string[] = ['']
-  let index = 0
+  const header = getHeader(name)
+  const overheadLength = header.length + wrapDiffCodeBlock('').length
 
-  for (const diff of diffs) {
-    const currentLength =
-      wrapDiffCodeBlock(messages[index]).length + header.length
-    const nextDiff = diffToString(diff)
+  const maxLength = MAX_MESSAGE_LENGTH - overheadLength
+  const messages = bundleMessages(
+    diffs.flatMap((diff) => contractDiffToMessages(diff, overheadLength)),
```

Comment:
```suggestion
    diffs.flatMap((diff) => contractDiffToMessages(diff, maxLength)),
```
I think that would be more self-explanatory if we pass here `maxLength` and then inside the function, we subtract `contractHeader.length` from it. That's nit picking tho